src/main/java/ch/njol/skript/conditions/CondEntityIsIn.java

```diff
@@ -0,0 +1,138 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.conditions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.conditions.base.PropertyCondition;
+import ch.njol.skript.doc.*;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.util.Kleenean;
+import org.bukkit.entity.Entity;
+import org.bukkit.event.Event;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Is in Rain/Lava/Water/BubbleColumn")
+@Description("Checks whether an entity is in Rain/Lava/Water/BubbleColumn")
+@Examples({"player is in rain",
+		"player is in water",
+		"player is in lava",
+		"player is in bubble column",
+		"player is in water or rain",
+		"player is in water or bubble column",
+		"player is in water or rain or bubble column"})
+@RequiredPlugins("Minecraft 1.16+, Paper 1.16+ (rain, lava and bubble column)")
+@Since("INSERT VERSION")
+public class CondEntityIsIn extends PropertyCondition<Entity> {
+	
+	static {
+		StringBuilder patterns = new StringBuilder("");
+		if (Skript.methodExists(Entity.class, "isInWater"))
```

Comment:
Do all of these method exists checks need to be here? Or does the existence of one method indicate that some others also exist? Can just help out with all the checking/appending :)